test/test_common/test_time_system.h

```diff
@@ -10,22 +10,33 @@
 namespace Envoy {
 namespace Event {
 
-class TestTimeSystem;
-
 // Adds sleep() and waitFor() interfaces to Event::TimeSystem.
 class TestTimeSystem : public Event::TimeSystem {
 public:
   ~TestTimeSystem() override = default;
 
   /**
    * Advances time forward by the specified duration, running any timers
-   * along the way that have been scheduled to fire.
+   * scheduled to fire, and blocking until the timer callbacks are complete.
+   * See also advanceTimeAsync(), which does not block.
    *
    * @param duration The amount of time to sleep.
    */
-  virtual void sleep(const Duration& duration) PURE;
-  template <class D> void sleep(const D& duration) {
-    sleep(std::chrono::duration_cast<Duration>(duration));
+  virtual void advanceTimeWait(const Duration& duration) PURE;
+  template <class D> void advanceTimeWait(const D& duration) {
+    advanceTimeWait(std::chrono::duration_cast<Duration>(duration));
+  }
+
+  /**
+   * Advances time forward by the specified duration. Timers may be triggered on
+   * their threads, but unlike advanceTimeWait(), this method does not block
+   * waiting for them to complete.
```

Comment:
Can you add some text to each function as to why a test writer would want to use each one? It may not be immediately clear.

Reply:
done: added: 
```
   * This function should be used in multi-threaded tests, where other
   * threads are running dispatcher loops. Integration tests should usually
   * use this variant.
```
and
```
   * This function should be used in single-threaded tests, in scenarios where
   * after time is advanced, the main test thread will run a dispatcher
   * loop. Unit tests will often use this variant.
```